encode.go

```diff
@@ -60,23 +59,16 @@ func newEncoder(writer io.Writer, opts *options) *encoder {
 		emitter: libyaml.NewEmitter(),
 	}
 
-	// Apply options if provided, otherwise use defaults
-	if opts != nil {
-		e.indent = opts.indent
-		e.lineWidth = opts.lineWidth
-		e.emitter.CompactSequenceIndent = opts.compactSeqIndent
-		e.emitter.SetWidth(opts.lineWidth)
-		e.emitter.SetUnicode(opts.unicode)
-		e.emitter.SetCanonical(opts.canonical)
-		e.emitter.SetLineBreak(opts.lineBreak)
-		e.explicitStart = opts.explicitStart
-		e.explicitEnd = opts.explicitEnd
-		e.flowSimpleCollections = opts.flowSimpleCollections
-	} else {
-		// Default values when no options provided
-		e.emitter.SetUnicode(true)
-		e.lineWidth = 80
-	}
+	e.indent = opts.indent
```

Comment:
Here and everywhere, I have a problem with ops being *options, so can be possibly nil

So here and at other place it can panic.

Otherwise, I agree with the change. But then maybe calling the Loader, Dump with yaml.V3 option could be considered instead of using legacy option (it might need a refactoring)

I'm busy with Christmas. I'll review in detail in a few days

Reply:
> Loader, Dump with yaml.V3 option

I think it would change the current behavior of Marshal (yaml.V3 has line wrapping).

Added a commit that cleans up the options internal a bit, and we now pass by value.